### PR TITLE
Java ice adapter

### PR DESCRIPTION
**Pull request by Geosearchef**
Not to be merged, contains a lot of changes for testing purposes.

### CODE REVIEW

**Comment by 1-alex98** [comment]
What exactly do you want us to do here?

**Comment by Geosearchef** [comment]
Take a look at the changes, they are WIP

File: build.gradle [hunk 1]
```diff
@@ -206,26 +205,14 @@ task githubRelease(type: ReleaseTask, overwrite: true) {
...
-  src "https://github.com/FAForever/ice-adapter/releases/download/v${faf_ice_adapter_version}/faf-ice-adapter-win32.zip"
-  dest file("${buildDir}/tmp/faf-ice-adapter.zip")
-  onlyIfNewer true
-}
 
 task downloadIceAdapterLinux(type: Download) {
   group "native dependencies"
-  src "https://github.com/FAForever/ice-adapter/releases/download/v${faf_ice_adapter_version}/faf-ice-adapter"
-  dest file("${buildDir}/resources/native/faf-ice-adapter")
+  src "https://geosearchef.de/faf-ice-adapter.jar"
```
Comment (1-alex98) [inline build.gradle]: I know you know but we need to change before merge

File: src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java [hunk 2]
```diff
@@ -93,27 +92,48 @@ void postConstruct() {
   }
 
   @SneakyThrows
-  private List<Map<String, String>> toIceServers(List<IceServersServerMessage.IceServer> iceServers) {
-    return iceServers.stream()
-        .map(this::toIceServer)
-        .collect(Collectors.toList());
-  }
+  private List<Map<String, Object>> toIceServers(List<IceServersServerMessage.IceServer> iceServers) {
+//    return iceServers.stream()
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java]: Remove comments

File: src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java [hunk 3]
```diff
@@ -93,27 +92,48 @@ void postConstruct() {
...
+
+    return (result);
   }
 
+//  @NotNull
+//  private Map<String, String> toIceServer(IceServersServerMessage.IceServer iceServer) {
+//    Map<String, String> map = new HashMap<>();
+//    map.put("url", iceServer.getUrl());
+//
+//    if (iceServer.getCredential() != null) {
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java]: Remove

File: src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java [hunk 4]
```diff
@@ -93,27 +92,48 @@ void postConstruct() {
...
+      map.put("urls", urls);
 
-    if (iceServer.getCredential() != null) {
       map.put("credential", iceServer.getCredential());
-      map.put("credentialType", iceServer.getCredentialType());
-    }
-    if (iceServer.getUsername() != null) {
-      map.put("username", iceServer.getUsername());
+      map.put("credentialType", "token");
+      map.put("username", iceServers.get(2).getUsername());
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java]: Why get(2)

File: src/main/resources/application-prod.yml [hunk 5]
```diff
@@ -1,26 +1,27 @@
 faf-client:
   server:
-    host: lobby.faforever.com
+    host: faf.micheljung.ch
+    port: 18001
 
   irc:
-    host: irc.faforever.com
+    host: irc.test.faforever.com
```
Comment (1-alex98) [inline src/main/resources/application-prod.yml]: Remove test...

File: src/main/resources/application-prod.yml [hunk 6]
```diff
@@ -1,26 +1,27 @@
...
-    remote-host: lobby.faforever.com
+    remote-host: test.faforever.com
 
   website:
-    base-url: https://www.faforever.com
+    base-url: https://www.test.faforever.com
 
   api:
-    base-url: https://api.faforever.com
+    base-url: http://api.faf.micheljung.ch:10080
```
Comment (1-alex98) [inline src/main/resources/application-prod.yml]: Reset

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**